### PR TITLE
Fix setup-go actions

### PR DESCRIPTION
**Pull request by silverwind**
The `setup-go` actions did not all have `check-latest` which means they use some cached version of go that currently still resolves to go1.20.4, as seen in a number of recent runs that currently fail at govulncheck because of it:

````
Run actions/setup-go@v4
Setup go version spec >=1.20
Attempting to resolve the latest version from the manifest...
matching >=1.20...
Resolved as '1.20.4'
````

Add the [check-latest](https://github.com/actions/setup-go#check-latest-version) option which should guarantee that this cache is skipped.

### CODE REVIEW

**Comment by silverwind** [comment]
Temporary change to backend files added to confirm that the actions now pass. Will revert before merge.

**Review by lunny — APPROVED** [review]

File: templates/home.tmpl [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {{template "base/head" .}}
+
```
Comment (wolfogre) [inline templates/home.tmpl]: Unrelated?

**Review by wolfogre — APPROVED** [review]

File: templates/home.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/home.tmpl]: See https://github.com/go-gitea/gitea/pull/25167#issuecomment-1584077734, it's just to trigger the backend pipelines which would otherwise be skipped because of the `files-changed` checks, will revert it after success.

**Comment by silverwind** [comment]
It still resolved to 1.20.4. I can only assume this is a bug in the action.

````
Setup go version spec >=1.20
Attempting to resolve the latest version from the manifest...
matching >=1.20...
Resolved as '1.20.4'

**Comment by silverwind** [comment]
Opened https://github.com/actions/setup-go/issues/384

**Comment by lunny** [comment]
Maybe the problems are in https://github.com/actions/go-versions/releases

**Comment by silverwind** [comment]
Let's merge it. Builds should automatically start to pass once https://github.com/actions/setup-go/issues/384 is resolved.
